exporter/newrelicexporter/newrelic.go

```diff
@@ -79,34 +79,44 @@ func newExporter(l *zap.Logger, c configmodels.Exporter) (*exporter, error) {
 }
 
 func (e exporter) pushTraceData(ctx context.Context, td pdata.Traces) (int, error) {
-	var errs []error
-	goodSpans := 0
-
-	octds := internaldata.TraceDataToOC(td)
-	for _, octd := range octds {
-
-		var srv string
-		if octd.Node != nil && octd.Node.ServiceInfo != nil {
-			srv = octd.Node.ServiceInfo.Name
-		}
-
-		transform := &transformer{
-			ServiceName: srv,
-			Resource:    octd.Resource,
+	var (
+		errs      []error
+		goodSpans int
+	)
+
+	for i := 0; i < td.ResourceSpans().Len(); i++ {
+		rspans := td.ResourceSpans().At(i)
+		if rspans.IsNil() {
+			continue
 		}
 
-		for _, span := range octd.Spans {
-			nrSpan, err := transform.Span(span)
-			if err != nil {
-				errs = append(errs, err)
+		resource := rspans.Resource()
+		for j := 0; j < rspans.InstrumentationLibrarySpans().Len(); j++ {
+			ispans := rspans.InstrumentationLibrarySpans().At(j)
+			if ispans.IsNil() {
 				continue
 			}
-			err = e.harvester.RecordSpan(nrSpan)
-			if err != nil {
-				errs = append(errs, err)
-				continue
+
+			transform := newTraceTransformer(resource, ispans.InstrumentationLibrary())
+			for k := 0; k < ispans.Spans().Len(); k++ {
+				span := ispans.Spans().At(k)
+				if span.IsNil() {
+					continue
+				}
+
+				nrSpan, err := transform.Span(span)
+				if err != nil {
+					// Record the error and try export anyway to help debug.
```

Comment:
What gets exported if the conversion failed? Partially transformed span? Seems a bit scary

Reply:
Yeah, we can probably just continue like before. The two situations where it would return a partial span (outside of a nil span) are a bad trace ID or bad span ID which the harvester will return an error:

https://github.com/newrelic/newrelic-telemetry-sdk-go/blob/e3d912863ace96681b2b66ebffdb0acc5f874ff2/telemetry/harvester.go#L106-L116

This means that nothing will be exported, but error will be duplicated.

I'll update.